app/models/mhv_account.rb

```diff
@@ -69,7 +73,7 @@ def terms_and_conditions_accepted?
   end
 
   def preexisting_account?
-    user&.mhv_correlation_id.present?
+    user&.mhv_correlation_id.present? && !previously_registered?
```

Comment:
I've redefined `preexisting_account?` to mean that the user has an `mhv_correlation_id` not from registering through Vets.gov. I noticed that this method hasn't been used anywhere, so I think the impact should be isolated to these changes.